flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/TableFormatFactoryBase.java

```diff
@@ -154,31 +156,37 @@ public static TableSchema deriveSchema(Map<String, String> properties) {
 
 		final TableSchema.Builder builder = TableSchema.builder();
 
-		final TableSchema baseSchema = TableSchemaUtils.getPhysicalSchema(
-			descriptorProperties.getTableSchema(SCHEMA));
-		for (int i = 0; i < baseSchema.getFieldCount(); i++) {
-			final String fieldName = baseSchema.getFieldNames()[i];
-			final DataType fieldType = baseSchema.getFieldDataTypes()[i];
+		final TableSchema tableSchema = descriptorProperties.getTableSchema(SCHEMA);
+		final TableSchema physicalSchema = TableSchemaUtils.getPhysicalSchema(tableSchema);
+
+		final Map<Integer, Integer> physicalIndices2Indices = Arrays.stream(physicalSchema.getFieldNames())
```

Comment:
A simper way to fix is add the field only when it is not generated. We already have `isGeneratedColumn `. So we don't need the complex mapping.